src/main/java/ch/njol/skript/registrations/EventValues.java

```diff
@@ -381,23 +390,52 @@ public static <T, E extends Event> Kleenean hasMultipleConverters(Class<E> event
 			return list;
 		// If the check should try again matching event values with a 0 time (most event values).
 		if (allowDefault && time != 0)
-			return getEventValueConverters(event, type, 0, false);
+			return getEventValueConverters(eventClass, valueClass, 0, false);
 		return null;
 	}
 
+	/*
+		In this method we can delegate/strip converters that are able to be obtainable through their own 'event-classinfo'.
+		For example, PlayerTradeEvent has a player value (player who traded) and an AbstractVillager value (villager traded from).
+		Beforehand, since there is no Entity value, it was then grabbing both values as they both can be casted as an Entity
+			resulting in a parse error of "multiple entities"
+		Now, we filter out the ones that can be obtained using their own classinfo, such as 'event-player'
+			which leaves us only the AbstractVillager for 'event-entity'
+	 */
```

Comment:
this should be javadocs

Reply:
I'm also confused on what delegation is going on here and why that's the verb used. It seems to me like it strips converts, it doesn't delegate them. I might be misreading, and if so i think whatever explanation there is that i'm missing should be added into the docs